drivers/sensor/ti/ina23x/ina230.c

```diff
@@ -20,21 +18,56 @@ LOG_MODULE_REGISTER(INA230, CONFIG_SENSOR_LOG_LEVEL);
 
 /** @brief The LSB value for the bus voltage register, in microvolts/LSB. */
 #define INA230_BUS_VOLTAGE_UV_LSB 1250U
+#define INA236_BUS_VOLTAGE_UV_LSB 1600U
 
 /** @brief The scaling for the power register. */
 #define INA230_POWER_SCALING 25
+#define INA236_POWER_SCALING 32
+
+/** @brief ADCRANGE bit in configuration register */
+#define INA236_CONFIG_ADC_RANGE BIT(12)
+
+/**
+ * @brief Defines new variant of the INA23X chip.
+ *
+ * @param BUS_VOLTAGE_UV_LSB	Resolution of 1 bit for bus voltage (1uV/bit).
+ * @param POWER_SCALING		The scaling for the power register.
+ * @param HAVE_HIGH_PRECISION	True if device supports high precision measurements.
+ */
+#define INA23X_DEFINE_TYPE(BUS_VOLTAGE_UV_LSB, POWER_SCALING, HAVE_HIGH_PRECISION)                 \
+	{                                                                                          \
+		.bus_voltage_uv_lsb = BUS_VOLTAGE_UV_LSB, .power_scaling = POWER_SCALING,          \
+		.have_high_precision = HAVE_HIGH_PRECISION                                         \
+	}
+
+enum ina23x_ids {
+	INA230,
+	INA236
+};
+
+struct ina23x_config {
+	uint16_t bus_voltage_uv_lsb;
+	uint8_t power_scaling;
+	bool have_high_precision;
+};
+
+static const struct ina23x_config ina23x_config[] = {
```

Comment:
There can be multiple instances of any combination of INA230 and INA236. Because of this, we need to know what configuration we need to apply so this array holds the configuration per device type and it is shared between instances of the same device type to save some memory.

Reply:
Without testing I guess it would require at least 3 devices to break even or save memory. Below 3 should cost more memory since you need the additional `void *`. The `void *` also increases code complexity IMO.

Reply:
Removed.